File: frontend/src/renderer/src/api/glossary.ts
// 术语库API服务
import { httpClient } from '@/utils/http'

// 术语库数据类型
export interface GlossaryCollection {
  id: number
  name: string
  description?: string
  icon?: string
  color?: string
  is_enabled: boolean
  term_count: number
  is_system: boolean
  created_at: string
  updated_at: string
}

export interface GlossaryCollectionCreate {
  name: string
  description?: string
  icon?: string
  color?: string
}

export interface GlossaryCollectionUpdate {
  name?: string
  description?: string
  icon?: string
  color?: string
  is_enabled?: boolean
}

export interface GlossaryCollectionListResponse {
  items: GlossaryCollection[]
  total: number
  page: number
  page_size: number
}

// 术语数据类型
export interface GlossaryTerm {
  id: number
  collection_id: number
  term: string
  synonyms: string[]
  description?: string
  is_enabled: boolean
  created_at: string
  updated_at: string
}

export interface GlossaryTermCreate {
  term: string
  synonyms: string[]
  description?: string
}

export interface GlossaryTermUpdate {
  term?: string
  synonyms?: string[]
  description?: string
  is_enabled?: boolean
}

export interface GlossaryTermListResponse {
  collection_id: number
  collection_name: string
  items: GlossaryTerm[]
  total: number
  page: number
  page_size: number
}

// 查询扩展相关类型
export interface MatchedTerm {
  id: number
  term: string
  synonyms: string[]
  collection_id: number
  collection_name: string
}

export interface GlossaryExpandRequest {
  query: string
  collection_ids?: number[]
}

export interface GlossaryExpandResponse {
  original_query: string
  matched_terms: MatchedTerm[]
  expanded_queries: string[]
  used_collections: string[]
}

// CSV导入响应
export interface GlossaryImportResponse {
  imported_count: number
  failed_count: number
  errors: Array<{
    row: number
    term: string
    error: string
  }>
}

// 术语扩展配置
export interface GlossaryExpansionConfig {
  enable: boolean
  collection_ids?: number[] | null
  max_expansion_terms?: number
}

export interface GlossaryExpansionConfigData {
  enable: boolean
  collection_ids?: number[] | null
  max_expansion_terms: number
  available_collections: Array<{
    id: number
    name: string
    description?: string
    icon?: string
    color?: string
    term_count: number
  }>
}

export interface GlossaryExpansionConfigResponse {
  success: boolean
  data: GlossaryExpansionConfigData
}

// 术语库服务
export class GlossaryService {
  // ========== 术语库管理 ==========

  // 获取术语库列表
  static async getCollections(page = 1, pageSize = 20, isEnabled?: boolean) {
    const params = new URLSearchParams()
    params.append('page', page.toString())
    params.append('page_size', pageSize.toString())
    if (isEnabled !== undefined) {
      params.append('is_enabled', isEnabled.toString())
    }

    return await httpClient.get(`/api/glossary/collections/?${params}`)
  }

  // 获取单个术语库
  static async getCollection(id: number) {
    return await httpClient.get(`/api/glossary/collections/${id}`)
  }

  // 创建术语库
  static async createCollection(data: GlossaryCollectionCreate) {
    return await httpClient.post('/api/glossary/collections/', data)
  }

  // 更新术语库
  static async updateCollection(id: number, data: GlossaryCollectionUpdate) {
    return await httpClient.put(`/api/glossary/collections/${id}`, data)
  }

  // 删除术语库
  static async deleteCollection(id: number) {
    return await httpClient.delete(`/api/glossary/collections/${id}`)
  }

  // 获取启用的术语库
  static async getEnabledCollections() {
    return await httpClient.get('/api/glossary/collections/enabled/list')
  }

  // ========== 术语管理 ==========

  // 获取术语列表
  static async getTerms(collectionId: number, page = 1, pageSize = 20, isEnabled?: boolean) {
    const params = new URLSearchParams()
    params.append('collection_id', collectionId.toString())
    params.append('page', page.toString())
    params.append('page_size', pageSize.toString())
    if (isEnabled !== undefined) {
      params.append('is_enabled', isEnabled.toString())
    }

    return await httpClient.get(`/api/glossary/terms/?${params}`)
  }

  // 获取单个术语
  static async getTerm(id: number) {
    return await httpClient.get(`/api/glossary/terms/${id}`)
  }

  // 创建术语
  static async createTerm(collectionId: number, data: GlossaryTermCreate) {
    const params = new URLSearchParams()
    params.append('collection_id', collectionId.toString())

    return await httpClient.post(`/api/glossary/terms/?${params}`, data)
  }

  // 更新术语
  static async updateTerm(collectionId: number, id: number, data: GlossaryTermUpdate) {
    const params = new URLSearchParams()
    params.append('collection_id', collectionId.toString())

    return await httpClient.put(`/api/glossary/terms/${id}?${params}`, data)
  }

  // 删除术语
  static async deleteTerm(collectionId: number, id: number) {
    const params = new URLSearchParams()
    params.append('collection_id', collectionId.toString())

    return await httpClient.delete(`/api/glossary/terms/${id}?${params}`)
  }

  // CSV导入
  static async importFromCSV(collectionId: number, file: File) {
    const formData = new FormData()
    formData.append('file', file)
    formData.append('collection_id', collectionId.toString())

    return await httpClient.post(`/api/glossary/terms/import?collection_id=${collectionId}`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  }

  // CSV导出
  static async exportToCSV(collectionId: number, locale: string = 'zh-CN') {
    const response = await httpClient.get(`/api/glossary/terms/export/${collectionId}`, {
      responseType: 'blob',
      headers: {
        'Accept-Language': locale
      }
    })

    // 创建下载链接
    const blob = new Blob([response as any], { type: 'text/csv' })
    const url = window.URL.createObjectURL(blob)
    const link = document.createElement('a')
    link.href = url
    link.download = `glossary_${collectionId}.csv`
    document.body.appendChild(link)
    link.click()
    document.body.removeChild(link)
    window.URL.revokeObjectURL(url)

    return response
  }

  // ========== 查询扩展 ==========

  // 扩展查询词
  static async expandQuery(query: string, collectionIds?: number[]) {
    return await httpClient.post('/api/glossary/expand', {
      query,
      collection_ids: collectionIds
    })
  }

  // ========== 术语扩展设置 ==========

  // 获取术语扩展配置
  static async getExpansionConfig(): Promise<GlossaryExpansionConfigResponse> {
    return await httpClient.get('/api/glossary/settings/')
  }

  // 更新术语扩展配置
  static async updateExpansionConfig(config: GlossaryExpansionConfig) {
    return await httpClient.post('/api/glossary/settings/', config)
  }
}

File: backend/app/schemas/glossary.py
# backend/app/schemas/glossary.py
"""
术语库相关的Pydantic模式定义
"""
from pydantic import BaseModel, Field, field_validator
from typing import Optional, List
from datetime import datetime


# ==================== 术语库Schema ====================

class GlossaryCollectionBase(BaseModel):
    """术语库基础模式"""
    name: str = Field(..., min_length=1, max_length=100, description="术语库名称")
    description: Optional[str] = Field(None, description="术语库描述")
    icon: Optional[str] = Field(None, max_length=50, description="图标（emoji）")
    color: Optional[str] = Field(None, max_length=20, description="颜色代码")


class GlossaryCollectionCreate(GlossaryCollectionBase):
    """创建术语库请求模式"""
    pass


class GlossaryCollectionUpdate(GlossaryCollectionBase):
    """更新术语库请求模式"""
    is_enabled: Optional[bool] = Field(None, description="是否启用")


class GlossaryCollectionResponse(GlossaryCollectionBase):
    """术语库响应模式"""
    id: int
    is_enabled: bool
    term_count: int
    is_system: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class GlossaryCollectionListResponse(BaseModel):
    """术语库列表响应模式"""
    items: List[GlossaryCollectionResponse]
    total: int
    page: int
    page_size: int


# ==================== 术语Schema ====================

class GlossaryTermBase(BaseModel):
    """术语基础模式"""
    term: str = Field(..., min_length=1, max_length=100, description="术语名称")
    synonyms: List[str] = Field(..., min_length=1, description="同义词列表")
    description: Optional[str] = Field(None, description="术语描述")
    examples: Optional[List[str]] = Field(None, description="示例用法")

    @field_validator('synonyms')
    @classmethod
    def validate_synonyms(cls, v):
        """验证同义词列表"""
        if not v or len(v) == 0:
            raise ValueError('同义词列表不能为空')
        # 去重
        return list(set(v))


class GlossaryTermCreate(GlossaryTermBase):
    """创建术语请求模式"""
    pass


class GlossaryTermUpdate(BaseModel):
    """更新术语请求模式"""
    term: Optional[str] = Field(None, min_length=1, max_length=100, description="术语名称")
    synonyms: Optional[List[str]] = Field(None, min_length=1, description="同义词列表")
    description: Optional[str] = Field(None, description="术语描述")
    examples: Optional[List[str]] = Field(None, description="示例用法")
    is_enabled: Optional[bool] = Field(None, description="是否启用")

    @field_validator('synonyms')
    @classmethod
    def validate_synonyms_optional(cls, v):
        """验证同义词列表（可选）"""
        if v is not None and len(v) == 0:
            raise ValueError('同义词列表不能为空')
        # 去重
        return list(set(v)) if v else v


class GlossaryTermResponse(GlossaryTermBase):
    """术语响应模式"""
    id: int
    collection_id: int
    is_enabled: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class GlossaryTermListResponse(BaseModel):
    """术语列表响应模式"""
    collection_id: int
    collection_name: str
    items: List[GlossaryTermResponse]
    total: int
    page: int
    page_size: int


# ==================== 查询扩展Schema ====================

class GlossaryExpandRequest(BaseModel):
    """查询扩展请求模式"""
    query: str = Field(..., min_length=1, description="查询词")
    collection_ids: Optional[List[int]] = Field(None, description="术语库ID列表，null表示全部")


class MatchedTerm(BaseModel):
    """匹配的术语"""
    id: int
    term: str
    synonyms: List[str]
    collection_id: int
    collection_name: str


class GlossaryExpandResponse(BaseModel):
    """查询扩展响应模式"""
    original_query: str
    matched_terms: List[MatchedTerm]
    expanded_queries: List[str]
    used_collections: List[str]


# ==================== CSV导入导出Schema ====================

class GlossaryImportResponse(BaseModel):
    """CSV导入响应模式"""
    imported_count: int = Field(..., description="成功导入数量")
    failed_count: int = Field(..., description="失败数量")
    errors: List[dict] = Field(default_factory=list, description="错误详情")


# ==================== 用户设置Schema ====================

class GlossarySettingsUpdate(BaseModel):
    """术语库设置更新模式"""
    enable_expansion: bool = Field(..., description="是否启用术语扩展")
    default_mode: str = Field(..., pattern="^(all|custom)$", description="默认模式：all或custom")
    selected_collections: Optional[List[int]] = Field(None, description="选中的术语库ID列表（仅custom模式）")
